### PR TITLE
[Merged by Bors] - feat: add FreeMonoid.lift_ofList

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/FreeMonoid/Basic.lean [hunk 1]
```diff
@@ -238,6 +237,11 @@ def lift : (α → M) ≃ (FreeMonoid α →* M)
 #align free_monoid.lift FreeMonoid.lift
 #align free_add_monoid.lift FreeAddMonoid.lift
 
+-- porting note: new
```
Comment (eric-wieser) [inline Mathlib/Algebra/FreeMonoid/Basic.lean:240]: I think we should maybe distinguish "added to make the port work" from "added after the port was complete"

Comment (urkud) [reply Mathlib/Algebra/FreeMonoid/Basic.lean:240]: How?

Comment (jcommelin) [reply Mathlib/Algebra/FreeMonoid/Basic.lean:240]: I think it doesn't matter too much. A simple `git blame` can tell whether it was added in a porting PR or in a "regular" PR.

Comment (eric-wieser) [reply Mathlib/Algebra/FreeMonoid/Basic.lean:240]: Only if we adopt some convention for marking in the git commit message; this isn't purely hypothetical, I'd like to be able to display mathlib4 porting commits on the port status page, and need a way to identify them

Comment (dupuisf) [reply Mathlib/Algebra/FreeMonoid/Basic.lean:240]: How about starting porting commit titles by "port:" instead of "feat:"?

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors r+

**Pull request closed without merging**